diff --git a/Cargo.lock b/Cargo.lock
index 208d5e3..d57acae 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -268,7 +268,7 @@ checksum = "cd7cc57abe963c6d3b9d8be5b06ba7c8957a930305ca90304f24ef040aa6f961"
 [[package]]
 name = "cloudflare"
 version = "0.12.0"
-source = "git+https://github.com/Wyn-Price/cloudflare-rs.git?branch=wyn/zone-details#a6179f8b3b520b17788f39fcd5f103e81a87a890"
+source = "git+https://github.com/Pablo1107/cloudflare-rs.git?branch=wyn/zone-details#59432045b1256ba46f66efc69fa3f152024ae180"
 dependencies = [
  "chrono",
  "http",
diff --git a/Cargo.toml b/Cargo.toml
index 4976bb6..5443ff6 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -19,7 +19,7 @@ version = "^4.4"
 features = ["wrap_help", "derive", "env"]
 
 [patch.crates-io]
-cloudflare = { git = "https://github.com/Wyn-Price/cloudflare-rs.git", branch = "wyn/zone-details" }
+cloudflare = { git = "https://github.com/Pablo1107/cloudflare-rs.git", branch = "wyn/zone-details" }
 public-ip = { git = "https://github.com/jcgruenhage/rust-public-ip.git", branch = "cloudflare-provider" }
 
 [profile.release]
diff --git a/src/dns.rs b/src/dns.rs
index 1fbbd18..ca74504 100644
--- a/src/dns.rs
+++ b/src/dns.rs
@@ -46,7 +46,6 @@ impl Clone_ for DnsRecord {
 			meta: Meta {
 				auto_added: self.meta.auto_added,
 			},
-			locked: self.locked,
 			ttl: self.ttl,
 			zone_id: self.zone_id.to_owned(),
 			modified_on: self.modified_on,
