diff --git a/src/nvd b/src/nvd
index 1cc429c..8cc63fc 100755
--- a/src/nvd
+++ b/src/nvd
@@ -1015,6 +1015,9 @@ def run_diff(
     if (left_resolved / "sw").is_dir() and (right_resolved / "sw").is_dir():
         left_selected_set = PackageSet.from_direct_dependencies((left_resolved / "sw").resolve())
         right_selected_set = PackageSet.from_direct_dependencies((right_resolved / "sw").resolve())
+    elif (left_resolved / "home-path").is_dir() and (right_resolved / "home-path").is_dir():
+        left_selected_set = PackageSet.from_direct_dependencies((left_resolved / "home-path").resolve())
+        right_selected_set = PackageSet.from_direct_dependencies((right_resolved / "home-path").resolve())
     else:
         left_selected_set = PackageSet.from_direct_dependencies(left_resolved)
         right_selected_set = PackageSet.from_direct_dependencies(right_resolved)
